INFO 2025-03-09 21:49:04,457 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 21:49:05,746 utils NumExpr defaulting to 8 threads.
DEBUG 2025-03-09 21:49:05,915 __init__ matplotlib data path: /Users/lzrong/anaconda3/lib/python3.11/site-packages/matplotlib/mpl-data
DEBUG 2025-03-09 21:49:05,918 __init__ CONFIGDIR=/Users/lzrong/.matplotlib
DEBUG 2025-03-09 21:49:05,919 __init__ interactive is False
DEBUG 2025-03-09 21:49:05,919 __init__ platform is darwin
DEBUG 2025-03-09 21:49:05,946 __init__ CACHEDIR=/Users/lzrong/.matplotlib
DEBUG 2025-03-09 21:49:05,947 font_manager Using fontManager instance from /Users/lzrong/.matplotlib/fontlist-v330.json
INFO 2025-03-09 21:49:07,256 firebase_client Firebase 憑證路徑: /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json
INFO 2025-03-09 21:49:07,257 firebase_client 憑證文件是否存在: False
ERROR 2025-03-09 21:49:07,257 firebase_client Firebase 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 21:49:07,257 firebase_client Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

ERROR 2025-03-09 21:49:07,257 views Error in post_list view: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 21:49:07,258 views Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 18, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 19, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

ERROR 2025-03-09 21:49:07,260 log Internal Server Error: /ptt/
Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 18, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 19, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stock_charts/views.py", line 12, in wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 43, in post_list
    return render(request, 'ptt_viewer/post_list.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 489, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 605, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 702, in __init__
    self.args_check(filter_name, filter_func, args)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 764, in args_check
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: default requires 2 arguments, 1 provided
INFO 2025-03-09 21:49:23,714 firebase_client Firebase 憑證路徑: /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json
INFO 2025-03-09 21:49:23,714 firebase_client 憑證文件是否存在: False
ERROR 2025-03-09 21:49:23,714 firebase_client Firebase 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 21:49:23,715 firebase_client Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

ERROR 2025-03-09 21:49:23,715 views Error in post_list view: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 21:49:23,715 views Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 18, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 19, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

ERROR 2025-03-09 21:49:23,721 log Internal Server Error: /ptt/
Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 18, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 19, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stock_charts/views.py", line 12, in wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 43, in post_list
    return render(request, 'ptt_viewer/post_list.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 489, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 605, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 702, in __init__
    self.args_check(filter_name, filter_func, args)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 764, in args_check
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: default requires 2 arguments, 1 provided
INFO 2025-03-09 21:49:24,552 firebase_client Firebase 憑證路徑: /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json
INFO 2025-03-09 21:49:24,552 firebase_client 憑證文件是否存在: False
ERROR 2025-03-09 21:49:24,552 firebase_client Firebase 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 21:49:24,553 firebase_client Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

ERROR 2025-03-09 21:49:24,553 views Error in post_list view: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 21:49:24,553 views Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 18, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 19, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

ERROR 2025-03-09 21:49:24,553 log Internal Server Error: /ptt/
Traceback (most recent call last):
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 18, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 19, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 36, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stock_charts/views.py", line 12, in wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 43, in post_list
    return render(request, 'ptt_viewer/post_list.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 489, in parse
    raise self.error(token, e)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 605, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 702, in __init__
    self.args_check(filter_name, filter_func, args)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 764, in args_check
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: default requires 2 arguments, 1 provided
INFO 2025-03-09 21:49:34,031 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py changed, reloading.
INFO 2025-03-09 22:16:14,964 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:16:28,696 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:16:34,089 basehttp "GET / HTTP/1.1" 200 2863
ERROR 2025-03-09 22:16:35,556 firebase_storage Firebase Storage 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:16:35,557 views 獲取股票列表時出錯: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
INFO 2025-03-09 22:16:35,560 basehttp "GET /mops/ HTTP/1.1" 200 3082
ERROR 2025-03-09 22:17:16,156 firebase_storage Firebase Storage 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:17:16,157 views 獲取股票列表時出錯: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
INFO 2025-03-09 22:17:16,164 basehttp "GET /mops/ HTTP/1.1" 200 3082
ERROR 2025-03-09 22:17:18,577 firebase_storage Firebase Storage 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:17:18,577 views 獲取股票列表時出錯: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
INFO 2025-03-09 22:17:18,582 basehttp "GET /mops/ HTTP/1.1" 200 3082
INFO 2025-03-09 22:17:19,937 basehttp "GET /ptt/ HTTP/1.1" 200 9118
ERROR 2025-03-09 22:17:21,108 firebase_storage Firebase Storage 初始化失敗: Storage bucket name not specified. Specify the bucket name via the "storageBucket" option when initializing the App, or specify the bucket name explicitly when calling the storage.bucket() function.
ERROR 2025-03-09 22:17:21,108 views 獲取股票列表時出錯: Storage bucket name not specified. Specify the bucket name via the "storageBucket" option when initializing the App, or specify the bucket name explicitly when calling the storage.bucket() function.
INFO 2025-03-09 22:17:21,109 basehttp "GET /mops/ HTTP/1.1" 200 3108
INFO 2025-03-09 22:17:50,079 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-09 22:17:50,727 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:18:02,576 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/mops_news/firebase_storage.py changed, reloading.
INFO 2025-03-09 22:18:03,515 autoreload Watching for file changes with StatReloader
ERROR 2025-03-09 22:18:14,094 firebase_storage Firebase Storage 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:18:14,095 views 獲取股票列表時出錯: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
INFO 2025-03-09 22:18:14,104 basehttp "GET /mops/ HTTP/1.1" 200 3082
ERROR 2025-03-09 22:18:16,150 firebase_storage Firebase Storage 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:18:16,150 views 獲取股票列表時出錯: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
INFO 2025-03-09 22:18:16,152 basehttp "GET /mops/ HTTP/1.1" 200 3082
INFO 2025-03-09 22:18:31,837 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/mops_news/firebase_storage.py changed, reloading.
INFO 2025-03-09 22:18:32,327 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:18:40,788 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py changed, reloading.
INFO 2025-03-09 22:18:41,364 autoreload Watching for file changes with StatReloader
ERROR 2025-03-09 22:18:44,836 firebase_storage Firebase Storage 初始化失敗: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:18:44,838 views 獲取股票列表時出錯: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
INFO 2025-03-09 22:18:44,852 basehttp "GET /mops/ HTTP/1.1" 200 3082
INFO 2025-03-09 22:18:53,917 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py changed, reloading.
INFO 2025-03-09 22:18:54,396 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:19:28,039 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py changed, reloading.
INFO 2025-03-09 22:19:29,496 autoreload Watching for file changes with StatReloader
ERROR 2025-03-09 22:20:41,927 log Internal Server Error: /ptt/
Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stock_charts/views.py", line 12, in wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/views.py", line 12, in post_list
    firebase_client = FirebasePTTClient.get_instance()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 14, in get_instance
    cls._instance = cls()
                    ^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py", line 27, in __init__
    cred = credentials.Certificate(cred_path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/firebase_admin/credentials.py", line 97, in __init__
    with open(cert) as json_file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stocksentiment-8cf69-firebase-adminsdk-rsalw-a998bda61f.json'
ERROR 2025-03-09 22:20:41,933 basehttp "GET /ptt/ HTTP/1.1" 500 83070
INFO 2025-03-09 22:22:42,501 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/mops_news/firebase_storage.py changed, reloading.
INFO 2025-03-09 22:22:43,062 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:22:50,770 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py changed, reloading.
INFO 2025-03-09 22:22:51,225 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:22:57,137 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-09 22:22:59,105 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-09 22:22:59,428 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 22:23:01,330 basehttp "GET /mops/1773/ HTTP/1.1" 200 3570
INFO 2025-03-09 22:23:03,421 basehttp "GET /mops/1773/1140226_173252/ HTTP/1.1" 200 3066
INFO 2025-03-09 22:23:06,300 basehttp "GET /mops/1773/ HTTP/1.1" 200 3570
INFO 2025-03-09 22:23:07,516 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 22:23:14,382 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/mops_news/firebase_storage.py changed, reloading.
INFO 2025-03-09 22:23:15,008 autoreload Watching for file changes with StatReloader
INFO 2025-03-09 22:24:39,579 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-09 22:24:39,925 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 22:24:41,223 basehttp "GET /mops/2330/ HTTP/1.1" 200 3178
INFO 2025-03-09 22:24:49,088 basehttp "GET /mops/2330/1111004_173546/ HTTP/1.1" 200 3066
INFO 2025-03-09 22:24:59,356 basehttp "GET /mops/2330/ HTTP/1.1" 200 3178
INFO 2025-03-09 22:25:01,613 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 22:25:18,977 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-09 22:25:32,707 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 22:25:34,969 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-09 22:25:36,843 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-09 22:25:37,905 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 22:58:27,268 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 22:58:42,410 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 22:58:44,020 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-09 22:58:45,297 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-09 22:58:46,069 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 22:58:47,138 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-09 22:58:47,827 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 22:58:48,405 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 23:06:37,629 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 23:06:38,671 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-09 23:06:39,925 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-09 23:06:41,370 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-09 23:06:50,395 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-09 23:15:31,945 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/ptt_viewer/firebase_client.py changed, reloading.
INFO 2025-03-09 23:15:32,908 autoreload Watching for file changes with StatReloader
INFO 2025-03-10 00:25:25,595 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-10 00:25:26,033 basehttp "GET /mops/1773/ HTTP/1.1" 200 3570
INFO 2025-03-10 00:25:27,240 basehttp "GET /mops/1773/1140226_173252/ HTTP/1.1" 200 3066
INFO 2025-03-10 00:25:31,405 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-10 00:25:36,095 basehttp "GET /ptt/post/-OKwvpEAnVKjUIn4QdTx/ HTTP/1.1" 200 51464
INFO 2025-03-10 00:27:21,987 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-10 00:27:23,294 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-10 00:27:24,538 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-10 00:27:25,502 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-10 01:36:07,274 basehttp "GET /ptt/post/-OKwvpFVc7Ps8pUADhEM/ HTTP/1.1" 200 18071
INFO 2025-03-10 18:59:58,719 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-10 19:00:01,214 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-10 19:00:02,513 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-10 19:00:06,382 basehttp "GET /stocks/?ticker=2330.tw&start_date=2024-09-11&end_date=2025-03-10&generate=1 HTTP/1.1" 200 224913
INFO 2025-03-10 19:00:24,217 basehttp "GET /stocks/?ticker=2330.tw&start_date=2024-09-11&end_date=2025-03-11&generate=1 HTTP/1.1" 200 225289
INFO 2025-03-11 03:21:46,097 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 03:21:47,112 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 03:23:16,632 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/urls.py changed, reloading.
INFO 2025-03-11 03:23:17,149 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 03:26:39,111 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 03:26:40,031 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 03:30:02,345 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 03:30:03,259 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 03:31:32,218 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 03:31:32,785 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 03:35:05,770 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 03:35:07,360 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 18:27:53,277 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 18:27:54,753 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-11 18:27:57,112 basehttp "GET /stocks/?ticker=2330.tw&start_date=2024-09-12&end_date=2025-03-11&generate=1 HTTP/1.1" 200 227549
INFO 2025-03-11 19:40:15,926 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 19:40:24,512 basehttp "GET /stocks/?ticker=2330.tw&start_date=2024-09-12&end_date=2025-03-11&generate=1 HTTP/1.1" 200 227549
INFO 2025-03-11 19:40:24,957 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 19:40:26,210 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-11 19:40:27,344 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-11 19:40:28,472 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-11 19:40:28,762 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-11 19:40:30,413 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:08:03,223 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:08:10,179 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:08:12,102 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:08:13,381 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-11 22:08:15,653 basehttp "GET /stocks/?ticker=2330.tw&start_date=2024-09-12&end_date=2025-03-11&generate=1 HTTP/1.1" 200 227545
INFO 2025-03-11 22:08:16,572 basehttp "GET /ptt/ HTTP/1.1" 200 9118
INFO 2025-03-11 22:08:17,364 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-11 22:08:17,638 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-11 22:13:30,463 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/urls.py changed, reloading.
INFO 2025-03-11 22:13:31,230 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:14:35,895 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-11 22:14:36,249 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-11 22:16:30,360 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py changed, reloading.
INFO 2025-03-11 22:16:30,863 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:16:44,330 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 22:16:44,764 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:18:37,544 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:18:44,426 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-11 22:18:44,827 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-11 22:18:47,484 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:18:50,087 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:18:50,757 basehttp "GET /stocks/ HTTP/1.1" 200 3955
INFO 2025-03-11 22:18:51,704 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:18:52,714 basehttp "GET /mops/ HTTP/1.1" 200 7134
INFO 2025-03-11 22:18:53,818 basehttp "GET / HTTP/1.1" 200 2863
INFO 2025-03-11 22:25:59,079 basehttp "GET / HTTP/1.1" 200 3833
ERROR 2025-03-11 22:26:00,556 log Internal Server Error: /notifications/
Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 134, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 131, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 109, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 163, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py", line 23, in notification_page
    return render(request, 'notifications/notification_page.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 109, in render
    reraise(exc, self.backend)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 130, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: base.html
ERROR 2025-03-11 22:26:00,557 basehttp "GET /notifications/ HTTP/1.1" 500 142495
ERROR 2025-03-11 22:26:35,162 log Internal Server Error: /notifications/
Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 134, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 131, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 109, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 163, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py", line 23, in notification_page
    return render(request, 'notifications/notification_page.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 109, in render
    reraise(exc, self.backend)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 130, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: base.html
ERROR 2025-03-11 22:26:35,169 basehttp "GET /notifications/ HTTP/1.1" 500 142632
INFO 2025-03-11 22:26:42,104 autoreload Watching for file changes with StatReloader
ERROR 2025-03-11 22:26:48,163 log Internal Server Error: /notifications/
Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 134, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 131, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader_tags.py", line 109, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/engine.py", line 163, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py", line 23, in notification_page
    return render(request, 'notifications/notification_page.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 109, in render
    reraise(exc, self.backend)
  File "/Users/lzrong/anaconda3/lib/python3.11/site-packages/django/template/backends/django.py", line 130, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: base.html
ERROR 2025-03-11 22:26:48,165 basehttp "GET /notifications/ HTTP/1.1" 500 142495
INFO 2025-03-11 22:26:59,150 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:27:20,994 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:27:39,056 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:28:23,217 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:28:25,279 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:28:27,726 basehttp "GET / HTTP/1.1" 200 3833
INFO 2025-03-11 22:28:30,275 basehttp "GET / HTTP/1.1" 200 3833
INFO 2025-03-11 22:28:31,605 firebase_storage Firebase Storage 初始化成功
INFO 2025-03-11 22:28:31,950 basehttp "GET /mops/ HTTP/1.1" 200 8104
INFO 2025-03-11 22:28:33,705 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:31:12,352 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py changed, reloading.
INFO 2025-03-11 22:31:13,019 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:31:19,859 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:31:26,888 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:31:58,069 basehttp "GET /notifications/ HTTP/1.1" 200 4636
INFO 2025-03-11 22:33:28,232 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:33:44,475 basehttp "GET /notifications/ HTTP/1.1" 200 4722
INFO 2025-03-11 22:33:51,300 basehttp "GET / HTTP/1.1" 200 3833
INFO 2025-03-11 22:33:52,862 basehttp "GET /notifications/ HTTP/1.1" 200 4722
INFO 2025-03-11 22:34:31,601 basehttp "GET /stocks/ HTTP/1.1" 200 4925
INFO 2025-03-11 22:34:36,255 basehttp "GET /ptt/ HTTP/1.1" 200 10088
INFO 2025-03-11 22:34:37,806 basehttp "GET /ptt/post/-OKwvpIDe5mZAh0kOf1A/ HTTP/1.1" 200 7612
INFO 2025-03-11 22:34:47,714 basehttp "GET /notifications/ HTTP/1.1" 200 4722
INFO 2025-03-11 22:36:49,484 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:37:32,887 basehttp "GET / HTTP/1.1" 200 3877
INFO 2025-03-11 22:37:36,779 basehttp "GET /notifications/ HTTP/1.1" 200 10154
INFO 2025-03-11 22:39:50,842 basehttp "GET /notifications/ HTTP/1.1" 200 10154
INFO 2025-03-11 22:39:52,284 basehttp "GET /notifications/ HTTP/1.1" 200 10154
INFO 2025-03-11 22:40:14,464 basehttp "GET /notifications/ HTTP/1.1" 200 15356
WARNING 2025-03-11 22:40:20,186 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:40:20,191 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
WARNING 2025-03-11 22:40:55,808 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:40:55,809 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
INFO 2025-03-11 22:41:12,448 basehttp "GET /notifications/ HTTP/1.1" 200 12382
WARNING 2025-03-11 22:41:12,500 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:41:12,500 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
WARNING 2025-03-11 22:41:14,056 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:41:14,056 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
WARNING 2025-03-11 22:41:19,484 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:41:19,484 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
INFO 2025-03-11 22:45:32,335 basehttp "GET /notifications/ HTTP/1.1" 200 13039
WARNING 2025-03-11 22:45:46,816 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 404 1889
INFO 2025-03-11 22:46:26,586 basehttp "GET /notifications/ HTTP/1.1" 200 13039
INFO 2025-03-11 22:46:28,295 basehttp "GET /notifications/ HTTP/1.1" 200 13039
WARNING 2025-03-11 22:46:39,712 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 404 1889
INFO 2025-03-11 22:49:27,558 basehttp "GET /notifications/ HTTP/1.1" 200 15874
WARNING 2025-03-11 22:50:12,296 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:50:12,301 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
WARNING 2025-03-11 22:50:22,139 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 404 1889
WARNING 2025-03-11 22:51:05,171 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:51:05,172 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
WARNING 2025-03-11 22:51:15,562 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 404 1889
INFO 2025-03-11 22:51:31,985 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/settings.py changed, reloading.
INFO 2025-03-11 22:51:32,997 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:51:49,403 basehttp "GET / HTTP/1.1" 200 3877
INFO 2025-03-11 22:51:50,543 basehttp "GET /notifications/ HTTP/1.1" 200 15874
WARNING 2025-03-11 22:51:52,705 log Not Found: /firebase-messaging-sw.js
WARNING 2025-03-11 22:51:52,705 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 404 3865
INFO 2025-03-11 22:51:52,748 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/urls.py changed, reloading.
INFO 2025-03-11 22:51:53,222 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:52:02,730 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 200 46
INFO 2025-03-11 22:52:17,516 basehttp "GET /notifications/ HTTP/1.1" 200 15874
INFO 2025-03-11 22:52:19,271 basehttp "GET /notifications/ HTTP/1.1" 200 15874
INFO 2025-03-11 22:52:20,800 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 1491
INFO 2025-03-11 22:52:20,837 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 1491
INFO 2025-03-11 22:52:30,812 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 22:52:33,807 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 22:53:45,356 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 22:53:51,464 basehttp "GET / HTTP/1.1" 200 3877
INFO 2025-03-11 22:53:52,790 basehttp "GET /notifications/ HTTP/1.1" 200 15065
INFO 2025-03-11 22:53:52,858 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 1491
INFO 2025-03-11 22:54:04,786 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 22:54:47,161 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 22:55:24,202 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 22:58:28,277 basehttp "GET /notifications/ HTTP/1.1" 200 18301
INFO 2025-03-11 22:58:28,354 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 1491
INFO 2025-03-11 22:58:47,481 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 22:59:31,169 basehttp "GET /notifications/ HTTP/1.1" 200 18301
INFO 2025-03-11 22:59:31,318 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3198
INFO 2025-03-11 22:59:44,015 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:00:31,443 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:02:10,739 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:04:48,998 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:05:36,600 basehttp "GET /notifications/ HTTP/1.1" 200 23010
INFO 2025-03-11 23:05:36,678 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3198
WARNING 2025-03-11 23:05:36,735 log Not Found: /notifications/subscribe_topic/
WARNING 2025-03-11 23:05:36,736 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 404 4367
WARNING 2025-03-11 23:05:47,728 log Not Found: /notifications/subscribe_topic/
WARNING 2025-03-11 23:05:47,734 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 404 4367
INFO 2025-03-11 23:05:49,010 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py changed, reloading.
INFO 2025-03-11 23:05:49,755 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 23:05:58,357 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/urls.py changed, reloading.
INFO 2025-03-11 23:05:58,799 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 23:07:06,496 basehttp "GET /notifications/ HTTP/1.1" 200 23010
INFO 2025-03-11 23:07:06,631 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:07:09,930 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:07:19,224 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:07:51,698 basehttp "GET /notifications/ HTTP/1.1" 200 23010
INFO 2025-03-11 23:07:51,761 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:10:04,221 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:10:13,689 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:10:29,463 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:10:29,576 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:10:31,549 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:10:31,597 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:15:40,211 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:15:40,317 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:15:45,192 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:20:58,230 basehttp "GET /notifications/subscribe_topic HTTP/1.1" 301 0
WARNING 2025-03-11 23:20:58,236 log Method Not Allowed (GET): /notifications/subscribe_topic/
WARNING 2025-03-11 23:20:58,236 basehttp "GET /notifications/subscribe_topic/ HTTP/1.1" 405 0
WARNING 2025-03-11 23:21:05,061 log Not Found: /subscribe_topic
WARNING 2025-03-11 23:21:05,062 basehttp "GET /subscribe_topic HTTP/1.1" 404 4205
WARNING 2025-03-11 23:21:11,958 log Method Not Allowed (GET): /notifications/subscribe_topic/
WARNING 2025-03-11 23:21:11,962 basehttp "GET /notifications/subscribe_topic/ HTTP/1.1" 405 0
INFO 2025-03-11 23:31:09,387 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:31:15,943 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:31:15,974 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:31:22,700 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:32:10,648 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:32:16,110 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:32:16,849 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:32:20,436 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:32:20,503 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:32:50,140 basehttp "GET / HTTP/1.1" 200 3877
INFO 2025-03-11 23:32:51,592 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:32:51,631 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:32:53,263 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:33:18,773 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:33:19,594 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:41:24,559 basehttp "GET / HTTP/1.1" 200 3877
INFO 2025-03-11 23:41:26,596 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:41:26,668 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:41:34,285 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:42:02,323 basehttp "GET /static/img/notification-icon.png HTTP/1.1" 304 0
INFO 2025-03-11 23:42:06,299 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:42:07,730 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:42:14,034 basehttp "GET /notifications/ HTTP/1.1" 200 16179
INFO 2025-03-11 23:42:14,111 basehttp "GET /firebase-messaging-sw.js HTTP/1.1" 200 3672
INFO 2025-03-11 23:42:16,246 basehttp "POST /notifications/subscribe_topic/ HTTP/1.1" 200 125
INFO 2025-03-11 23:47:38,831 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py changed, reloading.
INFO 2025-03-11 23:47:39,916 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 23:47:51,701 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/stockBots/urls.py changed, reloading.
INFO 2025-03-11 23:47:52,532 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 23:48:06,490 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py changed, reloading.
INFO 2025-03-11 23:48:06,924 autoreload Watching for file changes with StatReloader
INFO 2025-03-11 23:48:10,876 autoreload /Users/lzrong/PycharmProjects/write_share_robots/課程範例程式/ch23/stockBots/notifications/views.py changed, reloading.
INFO 2025-03-11 23:48:11,358 autoreload Watching for file changes with StatReloader
